servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -0,0 +1,455 @@
+      }
+    } catch (IOException e) {
+      throw new FailedToCheckpointException("Failed to rename " + srcPath + " to " + dstPath);
+    }
+    long fileSize;
+    try {
+      fileSize = mUfs.getFileSize(dstPath);
+    } catch (IOException e) {
+      throw new FailedToCheckpointException("Failed to getFileSize " + dstPath);
+    }
+    mMasterClient.addCheckpoint(mWorkerId, fileId, fileSize, dstPath);
+  }
+
+  /**
+   * Cleans up after users, to prevent zombie users. This method is called periodically.
+   */
+  public void cleanupUsers() throws IOException {
+    for (long user : mUsers.getTimedOutUsers()) {
+      mUsers.removeUser(user);
+      mBlockStore.cleanupUser(user);
+    }
+  }
+
+  /**
+   * Commits a block to Tachyon managed space. The block must be temporary. The block is
+   * persisted after {@link BlockStore#commitBlock(long, long)}. The block will not be accessible
+   * until {@link MasterClient#worker_cacheBlock(long, long, long, long, long)}
+   * succeeds
+   *
+   * @param userId The id of the client
+   * @param blockId The id of the block to commit
+   * @return true if successful, false otherwise
+   * @throws IOException if the block to commit does not exist
+   */
+  // TODO: This may be better as void
+  public boolean commitBlock(long userId, long blockId) throws IOException {
+    mBlockStore.commitBlock(userId, blockId);
+
+    // TODO: Reconsider how to do this without heavy locking
+    // Block successfully committed, update master with new block metadata
+    Long lockId = mBlockStore.lockBlock(userId, blockId);
+    BlockMeta meta = mBlockStore.getBlockMeta(userId, blockId, lockId);
+    BlockStoreLocation loc = meta.getBlockLocation();
+    Long storageDirId = loc.getStorageDirId();
+    Long length = meta.getBlockSize();
+    BlockStoreMeta storeMeta = mBlockStore.getBlockStoreMeta();
+    Long bytesUsedOnTier = storeMeta.getUsedBytesOnTiers().get(loc.tierLevel());
+    try {
+      mMasterClient
+          .worker_cacheBlock(mWorkerId, bytesUsedOnTier, storageDirId, blockId, length);
+    } catch (TException te) {
+      mBlockStore.unlockBlock(userId, blockId);
+      throw new IOException("Failed to commit block to master.", te);
+    }
+    mBlockStore.unlockBlock(userId, blockId);
+    return true;
+  }
+
+  /**
+   * Creates a block in Tachyon managed space. The block will be temporary until it is committed.
+   *
+   * @param userId The id of the client
+   * @param blockId The id of the block to create
+   * @param location The tier to place the new block in, -1 for any tier
+   * @param initialBytes The initial amount of bytes to be allocated
+   * @return A string representing the path to the local file
+   * @throws IOException if the block already exists
+   * @throws OutOfSpaceException if there is no more space to store the block
+   */
+  // TODO: We should avoid throwing IOException
+  public String createBlock(long userId, long blockId, int location, long initialBytes)
```

Comment:
What `location` means here is not clear, maybe rename it to tierAlias? Or you may explain what location means in javadoc more explicitly.